hare/messagevalidation.go

```diff
@@ -32,9 +34,17 @@ func (ev *eligibilityValidator) validateRole(m *Msg) bool {
 	// TODO: validate role proof sig
 
 	pub := m.PubKey
+	layer := types.LayerID(m.InnerMsg.InstanceId)
+	nId := types.NodeId{Key: pub.String()}
+
 	// validate role
-	if !ev.oracle.Eligible(InstanceId(m.InnerMsg.InstanceId), m.InnerMsg.K, pub.String(), Signature(m.InnerMsg.RoleProof)) {
-		ev.Warning("Role validation failed")
+	res, err := ev.oracle.Eligible(layer, m.InnerMsg.K, expectedCommitteeSize(m.InnerMsg.K, ev.n), nId, m.InnerMsg.RoleProof)
+	if err != nil {
+		ev.Error("Could not retrieve eligibility result err=%v", err)
+		return false
```

Comment:
It is possible that the error was caused due to an internal bug. In such case reporting that the node isn't eligible is wrong, consider the case where we blacklist a node because it wasn't eligible only due to an internal bug. `validateRole` should return an error.

Reply:
Fixed